### PR TITLE
Fix Valhalla Jtreg StrictFields test

### PR DESCRIPTION
**Pull request by theresa-m**
- Only recognize strict field for valuetype class versions
- Remove outdated value type field verification checks (@hangshao0 can you confirm this? I wasn't able to track down what draft spec these were part of and I can't find anything in the latest version)

Related to https://github.com/eclipse-openj9/openj9/pull/19830 and https://github.com/eclipse-openj9/openj9/issues/13182

### CODE REVIEW

File: runtime/bcutil/cfreader.c [hunk 1]
```diff
@@ -1708,35 +1708,25 @@ checkFields(J9PortLibrary* portLib, J9CfrClassFile * classfile, U_8 * segment, U
...
 		}
 
-		/* A field must not have set both ACC_STRICT and ACC_STATIC. */
-		if (J9_ARE_ALL_BITS_SET(value, CFR_ACC_STRICT | CFR_ACC_STATIC)) {
-			errorCode = J9NLS_CFR_ERR_FIELD_CANT_BE_STRICT_AND_STATIC__ID;
-			goto _errorFound;
-		}
+		if (J9_IS_CLASSFILE_OR_ROMCLASS_VALUETYPE_VERSION(classfile)) {
+			/* A field must not have set both ACC_STRICT and ACC_STATIC. */
+			if (J9_ARE_ALL_BITS_SET(value, CFR_ACC_STRICT | CFR_ACC_STATIC)) {
```
Comment (hangshao0) [inline runtime/bcutil/cfreader.c:1720]: You can format the code as:
```
if (J9_ARE_ALL_BITS_SET(value, CFR_ACC_STRICT) {
    if (J9_ARE_ALL_BITS_SET(value, CFR_ACC_STATIC) {
      /* error case */
    }
    if (J9_ARE_NO_BITS_SET(value, CFR_ACC_FINAL) {
       /* error case */
    }
}
```

File: runtime/bcutil/cfreader.c [hunk 2]
```diff
@@ -1708,35 +1708,25 @@ checkFields(J9PortLibrary* portLib, J9CfrClassFile * classfile, U_8 * segment, U
 
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 		if (J9_IS_CLASSFILE_VALUETYPE(classfile)) {
-			if (J9_ARE_ALL_BITS_SET(classfile->accessFlags, CFR_ACC_ABSTRACT)) {
-				if (J9_ARE_NO_BITS_SET(value, CFR_ACC_STATIC)) {
-					errorCode = J9NLS_CFR_ERR_MISSING_ACC_STATIC_ON_ABSTRACT_IDENTITYLESS_CLASS_FIELD__ID;
-					goto _errorFound;
-				}
-			} else {
-				if (J9_ARE_NO_BITS_SET(value, CFR_ACC_STATIC | CFR_ACC_FINAL)) {
```
Comment (hangshao0) [inline runtime/bcutil/cfreader.c:1717]: There should be a check that each field of a value class must have exactly one of its `ACC_STATIC` or `ACC_STRICT `flags set.

File: runtime/bcverify/vrfyhelp.c [hunk 3]
```diff
@@ -906,7 +906,10 @@ isFieldAccessCompatible(J9BytecodeVerificationData *verifyData, J9ROMFieldRef *f
 		J9BranchTargetStack *liveStack = (J9BranchTargetStack *)verifyData->liveStack;
 		J9ROMFieldShape *field = findFieldFromCurrentRomClass(romClass, fieldRef);
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-		IDATA isStrictField = (NULL != field) && J9_ARE_ALL_BITS_SET(field->modifiers, J9AccStrict);
+		IDATA isStrictField =
+			(NULL != field)
+			&& J9_IS_CLASSFILE_OR_ROMCLASS_VALUETYPE_VERSION(romClass)
```
Comment (hangshao0) [inline runtime/bcverify/vrfyhelp.c:911]: I suggest to add a macro `J9ROMFIELD_IS_CONTENDED` and include the class version check in the macro.

File: runtime/bcutil/cfreader.c (hunk 2 above)
Comment (hangshao0) [reply runtime/bcutil/cfreader.c:1717]: OK. I see the check below.

File: runtime/bcverify/vrfyhelp.c (hunk 3 above)
Comment (theresa-m) [reply runtime/bcverify/vrfyhelp.c:911]: Where is the language `IS_CONTENDED` used for strict fields?

Comment (hangshao0) [reply runtime/bcverify/vrfyhelp.c:911]: Sorry about the typo. `J9ROMFIELD_IS_CONTENDED` -> `J9ROMFIELD_IS_STRICT`

File: runtime/oti/j9.h [hunk 4]
```diff
@@ -352,6 +352,7 @@ static const struct { \
 #define J9_VALUETYPE_FLATTENED_SIZE(clazz) (J9CLASS_HAS_4BYTE_PREPADDING((clazz)) ? ((clazz)->totalInstanceSize - sizeof(U_32)) : (clazz)->totalInstanceSize)
 #define J9_IS_J9ARRAYCLASS_NULL_RESTRICTED(clazz) J9_ARE_ALL_BITS_SET((clazz)->classFlags, J9ClassArrayIsNullRestricted)
 #define J9CLASS_GET_NULLRESTRICTED_ARRAY(clazz) (J9_IS_J9CLASS_VALUETYPE(clazz) ? (clazz)->nullRestrictedArrayClass : NULL)
+#define J9ROMFIELD_IS_STRICT(romClassOrClassfile, fieldModifiers) J9_IS_CLASSFILE_OR_ROMCLASS_VALUETYPE_VERSION(romClassOrClassfile) && J9_ARE_ALL_BITS_SET(fieldModifiers, J9AccStrict)
```
Comment (hangshao0) [inline runtime/oti/j9.h:355]: `J9_IS_CLASSFILE_OR_ROMCLASS_VALUETYPE_VERSION(romClassOrClassfile) && J9_ARE_ALL_BITS_SET(fieldModifiers, J9AccStrict)` should be put inside `( )`

**Comment by hangshao0** [comment]
Jenkins test sanity.functional,extended alinuxval jdknext

**Comment by hangshao0** [comment]
Jenkins test sanity.functional,extended xlinuxval jdknext

**Review by hangshao0 — APPROVED** [review]

**Pull request merged**